### PR TITLE
Design doc

### PR DESCRIPTION
**Pull request by lum1n0us**

### CODE REVIEW

**Review by BenSunWhiteBoard — APPROVED** [review]
Looks great! I will add some materials and work on some TODOs with the information I gathered before.

File: docs/design.md [hunk 1]
```diff
@@ -1 +1,585 @@
 # how to implement a python binding of WAMR
+
+A python embedding of a Wasm runtime allows users to call a set of runtime's
+APIs, which is implemented by C, C++, or Rust, from the python world.
+
+In WAMR case, interfaces in `core/iwasm/include/wasm_c_api.h` will be able to
+be called from the python world. To achive that, we will create two stuffs:
```
Comment (xujuntwt95329) [inline docs/design.md]:
```suggestion
be called from the python world. To achieve that, we will create two stuffs:
```

**Comment by lum1n0us** [comment]
@wenyongh need your opinions 😃

File: docs/design.md [hunk 2]
```diff
@@ -1 +1,610 @@
 # how to implement a python binding of WAMR
+
+A python embedding of a Wasm runtime allows users to call a set of runtime's
+APIs, which is implemented by C, C++, or Rust, from the python world.
```
Comment (wenyongh) [inline docs/design.md]: Should be `which are implemented`?

File: docs/design.md [hunk 3]
```diff
@@ -1 +1,610 @@
...
+> data between Python and C. Other options include _cffi_, _pybind11_,
+> _cpython_ and so on. Because we tend to make the binding depending on least
+> items. The built-in module, _ctypes_, is a good choice.
+
+## General rules to marshal
+
+### load libraries
+
+The `ctypes` supports locating a dynamic link library in a way similar to the
+compiler does and loading the library with or without the path. In WAMR case,
```
Comment (wenyongh) [inline docs/design.md]: to the compiler does => to what the compiler does

File: docs/design.md [hunk 4]
```diff
@@ -1 +1,610 @@
...
+| wasm_memory_pages_t | c_uint32 |
+| wasm_mutability_t   | c_bool   |
+| wasm_table_size_t   | c_uint32 |
+| wasm_valkind_t      | c_uint8  |
+
+`ctypes.POINTER(datatype_name)` represents any kind of pointer.
+
+### structured datatypes
+
+To create a corresponding concept of each native structured data type, includes
```
Comment (wenyongh) [inline docs/design.md]: includes => including or which includes?

**Comment by lum1n0us** [comment]
thanks for catching. Has run a basic grammar check on grammarly.

**Pull request merged**